css/src/components/badge.scss

```diff
@@ -71,10 +76,34 @@ $badge-border-radius: $border-radius-rounded !default;
 		}
 	}
 
-	.icon {
+	&.badge-icon {
```

Comment:
Do we need badge-icon? What's the reasoning behind adding it? I know we often namespace subcomponents in this fashion, but in the past we've made a few exceptions, include for icon (see button's styling for examples).

Could we instead
1. target only children .icons to ensure they have their padding removed?
2. Set font-size at the top level. Then ...
3. If possible icon should have one `em` unit font sizing to ensure correct scaling across sizes.

```
.icon:only-child {
	padding: 0;
}
```

```
// 2 / 3

.badge {
  font-size: 1rem; // use var
  
  .icon {
     font-size: 1.2em; // if possible set an em value only once for the scaling of the icon with size classes
   }
  
  .badge-xs {
    font-size: .875rem; // or whatever value
    
    // if required
    .icon {
       font-size: somevalue;
   }
}

Reply:
It looks like `1.25em` is the right font size from what I can tell from the figma

Reply:
updated styles, added `icon-<size>`, this should be better